src/extension_utils.c

```diff
@@ -172,3 +176,56 @@ extension_current_state()
 
 	return EXTENSION_STATE_NOT_INSTALLED;
 }
+
+static void
+extension_load_without_preload()
+{
+	/* cannot use GUC variable here since extension not yet loaded */
+	char	   *allow_install_without_preload = GetConfigOptionByName("timescaledb.allow_install_without_preload", NULL, true);
+
+	if (allow_install_without_preload == NULL ||
+		strcmp(allow_install_without_preload, "on") != 0)
+	{
+		/*
+		 * These are FATAL because otherwise the loader ends up in a weird
+		 * half-loaded state after an ERROR
+		 */
+		/* Only privileged users can get the value of `config file` */
+#if PG10
+		if (is_member_of_role(GetUserId(), DEFAULT_ROLE_READ_ALL_SETTINGS))
+#else
+		if (superuser())
+#endif
+		{
+			char	   *config_file = GetConfigOptionByName("config_file", NULL, false);
+
+			ereport(FATAL,
+					(errmsg("The timescaledb library is not preloaded"),
```

Comment:
If we are changing this code, can we also update the error messages to conform to recommendations?

https://www.postgresql.org/docs/10/static/error-style-guide.html

Reply:
Btw, I think we should do this for the whole loader ASAP, since we want to avoid making changes too often. Might make sense in separate commit or PR.

Reply:
Added a commit